File: src/main/java/com/github/kindrat/cassandra/client/ui/editor/filter/PartialTableCondition.java
package com.github.kindrat.cassandra.client.ui.editor.filter;

import java.util.Optional;
import java.util.Set;

public class PartialTableCondition implements StateCondition {
    @Override
    public boolean isCurrentState(String[] words, Set<String> columnNames) {
        Optional<String> lastWord = lastWord(words);
        return lastWord.isPresent() && columnStartsWithWord(columnNames, lastWord.get());
    }

    @Override
    public State name() {
        return State.PARTIAL_TABLE;
    }

    private boolean columnStartsWithWord(Set<String> columnNames, String lastWord) {
        return columnNames.stream().anyMatch(s -> s.startsWith(lastWord) && !s.equals(lastWord));
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/filter/condition/PartialTableCondition.kt
package com.github.kindrat.cassandra.client.filter.condition

import com.github.kindrat.cassandra.client.util.StringUtil

class PartialTableCondition : StateCondition {

    override fun isCurrentState(words: Array<String>, columnNames: Set<String>): Boolean {
        val lastWord = StringUtil.lastWord(words)
        return lastWord?.let { word: String -> columnStartsWithWord(columnNames, word)} ?: false
    }

    override fun name(): StateCondition.State {
        return StateCondition.State.PARTIAL_TABLE
    }

    private fun columnStartsWithWord(columnNames: Set<String>, lastWord: String): Boolean {
        return columnNames.any { it.startsWith(lastWord) && it != lastWord }
    }
}